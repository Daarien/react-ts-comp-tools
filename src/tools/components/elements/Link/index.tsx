import { Component, ReactNode } from 'react';
import classnames from 'classnames';

const STATUS = {
  HOVERED: 'hovered',
  NORMAL: 'normal',
};

interface Props {
  page: string;
  children: ReactNode;
}

interface State {
  class: string;
}

export default class Link extends Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = {
      class: STATUS.NORMAL,
    };
  }

  _onMouseEnter = () => {
    this.setState({ class: STATUS.HOVERED });
  };

  _onMouseLeave = () => {
    this.setState({ class: STATUS.NORMAL });
  };

  render() {
    return (
      <a
        className={classnames(this.state.class, {
          ['active']: this.state.class === STATUS.HOVERED,
        })}
        href={this.props.page || '#'}
        onMouseEnter={this._onMouseEnter}
        onMouseLeave={this._onMouseLeave}
      >
        {this.props.children}
      </a>
    );
  }
}
