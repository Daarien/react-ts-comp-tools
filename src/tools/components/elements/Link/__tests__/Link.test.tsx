import Link from '..';
import renderer from 'react-test-renderer';

test('Link changes the class when hovered', () => {
  const component = renderer.create(<Link page="https://vk.com">Vkontakte</Link>);
  let tree: any = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree!.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //manually trigger the callback
  tree!.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
