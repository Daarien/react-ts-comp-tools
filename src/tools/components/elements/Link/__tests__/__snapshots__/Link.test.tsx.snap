// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://vk.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Vkontakte
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered active"
  href="https://vk.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Vkontakte
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://vk.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Vkontakte
</a>
`;
